solutions/cses-1650.mdx

```diff
@@ -2,41 +2,116 @@
 id: cses-1650
 source: CSES
 title: Range Xor Queries
-author: Danh Ta Chi Thanh
+author: Danh Ta Chi Thanh, Nathan Gong
 ---
 
-## Solution
+## Explanation
 
-[Unofficial Editorial](https://codeforces.com/blog/entry/77128?#comment-637536)
+Whenever we come across range queries, prefix sums should immediately come to
+mind. However, since we are dealing with range XOR, we have to use a prefix XOR
+array rather than a prefix sum array. To do so, we can define $\texttt{prefix}$
+as an array such that
+$\texttt{prefix}[i]= \texttt{arr}[0] \oplus \texttt{arr}[1] \oplus \dots \oplus \texttt{arr}[i]$.
+Thus, for each query $(a, b)$, our answer is simply
+$\texttt{prefix}[a-1]\oplus \texttt{prefix}[b]$.
+
+### Proof
+
+We know that
+$\texttt{prefix}[a-1]= \texttt{arr}[0] \oplus \texttt{arr}[1] \oplus \dots \oplus \texttt{arr}[a-1]$.
+We also know that
+$\texttt{prefix}[b]= \texttt{arr}[0] \oplus \texttt{arr}[1] \oplus \dots \oplus \texttt{arr}[b]$.
+
+Therefore:
```

Comment:
```suggestion
Therefore, since $a \oplus a = 0$:
```

Reply:
I think using `a` as a number might be misleading because it was already used in the explanation. I'll create another commit for this

Reply:
```suggestion
Therefore, since $x\oplus x = 0$ for any $x$:
```